### PR TITLE
feat(condo): DOMA-9373 change billing receipt notification flow

### PR DESCRIPTION
**Pull request by vovaaxeapolla**

### CODE REVIEW

File: apps/condo/domains/common/utils/date.js [hunk 1]
```diff
@@ -10,7 +10,7 @@ const DATE_FORMAT_Z = 'YYYY-MM-DDTHH:mm:ss.SSSZZ'
  * @param shouldFormat
  * @returns {*}
  */
-const getMonthStart = (dateRaw, shouldFormat = false) => {
+const getMonthStart = (dateRaw = undefined, shouldFormat = false) => {
```
Comment (sitozzz) [inline apps/condo/domains/common/utils/date.js:13]: What's the reason of passing undefined to `dateRaw` here?

File: apps/condo/domains/resident/tasks/helpers/sendBillingReceiptNotifications.spec.js [hunk 2]
```diff
@@ -18,9 +13,7 @@ describe('sendBillingReceiptNotifications', () => {
     describe('feature flag', () => {
         it('checks for proper result on enabled', async () => {
             setAllFeatureFlags(true)
-            const redisClient = getRedisClient()
-            await redisClient.set(REDIS_LAST_DATE_KEY, dayjs().toISOString())
-            expect(await sendBillingReceiptNotifications()).toBeUndefined()
+            expect(await sendBillingReceiptNotifications()).toEqual(undefined)
```
Comment (sitozzz) [inline apps/condo/domains/resident/tasks/helpers/sendBillingReceiptNotifications.spec.js:16]: If I'm correct you can use here `toBeUndefined()`
```suggestion
            expect(await sendBillingReceiptNotifications())..toBeUndefined()
```

File: apps/condo/domains/resident/tasks/helpers/sendBillingReceiptsAddedNotifications.js [hunk 3]
```diff
@@ -1,226 +1,31 @@
...
-     */
-    const lastDt = resendFromDt ? resendFromDt.replace(' ', 'T') : await redisClient.get(REDIS_LAST_DATE_KEY) || thisMonthStart
+const sendBillingReceiptsAddedNotifications = async () => {
+    logger.info({ msg: 'Starting billing receipts notification process' })
+    const lastSendDate = dayjs().subtract(1, 'day').toISOString()
+    const BillingContexts = await find('BillingIntegrationOrganizationContext', {
+        updatedAt_gte: lastSendDate,
+        status: CONTEXT_FINISHED_STATUS,
+        deletedAt: null,
+    })
```
Comment (sitozzz) [inline apps/condo/domains/resident/tasks/helpers/sendBillingReceiptsAddedNotifications.js:18]: What if there is more BillingContexts than default limit?

File: apps/condo/domains/common/utils/date.js (hunk 1 above)
Comment (vovaaxeapolla) [reply apps/condo/domains/common/utils/date.js:13]: This func is often called without any args and linter shows warnings. Undefined is normal value for it

File: apps/condo/domains/resident/tasks/sendBillingReceiptsAddedNotificationForOrganizationContextTask.js [hunk 4]
```diff
@@ -0,0 +1,257 @@
+const dayjs = require('dayjs')
+const get = require('lodash/get')
+const groupBy = require('lodash/groupBy')
+const isEmpty = require('lodash/isEmpty')
```
[the remark this answers is not in the document]
Comment (vovaaxeapolla) [reply apps/condo/domains/resident/tasks/sendBillingReceiptsAddedNotificationForOrganizationContextTask.js:4]: Ok

File: apps/condo/domains/resident/tasks/helpers/sendBillingReceiptNotifications.spec.js (hunk 2 above)
Comment (vovaaxeapolla) [reply apps/condo/domains/resident/tasks/helpers/sendBillingReceiptNotifications.spec.js:16]: Yes you're right

**Review by dkoviazin — APPROVED** [review]

**Review by abshnko — APPROVED** [review]

**Pull request merged**